common/components/ui/Modal.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import closeIcon from 'assets/images/icon-x.svg';
+
+import './Modal.scss';
+
+export default class Modal extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large', 'xlarge']),
```

Comment:
the max-width / width 100% combo makes it so that a container is at most max-width px, but will allow itself to shrink down to the size of its container. It's how most responsive things do what they do.

Unfortunately it doesn't have the same behavior if I do that, as the screenshot above shows. The `width: 100%;` doesn't work because the modal has `width: auto;` (To determine width from its content.) So the only thing that would work, is if the content set an explicit width. But then it wouldn't be responsive.